src/app/Login/Login.tsx

```diff
@@ -74,11 +85,10 @@ export const Login = (props) => {
   }, [serviceContext, serviceContext.api, addSubscription, onLoginSuccess, notifications, authMethod]);
 
   const handleSubmit = React.useCallback((evt, token, authMethod) => {
-    setToken(token);
     setAuthMethod(authMethod);
     checkAuth(token, authMethod, true);
     evt.preventDefault();
-  }, [setToken, setAuthMethod, checkAuth]);
+  }, [setAuthMethod, checkAuth]);
 
   React.useEffect(() => {
     const sub = serviceContext.api.getAuthMethod().subscribe(setAuthMethod);
```

Comment:
How does the service api remember token values? When I click `logout`, the sessionStorage token is removed but `checkAuth()` gets called after `api.service.getToken()`, which then automatically logs the user back into Cryostat.

I found `this.token.next(token);` in `Api.service.tsx` and am not sure how that gets saved to `private readonly token = new ReplaySubject<string>(1);`

Reply:
It looks like `ReplaySubject` will remember values for as long as the timeout (default is infinity).
https://rxjs-dev.firebaseapp.com/api/index/class/ReplaySubject 

Maybe we could create some sort of reset function?
https://stackoverflow.com/questions/51145664/resetting-replaysubject-in-rxjs-6/51147023#51147023

Reply:
Why does the `ReplaySubject` token have no expiration timeout?Instead of a reset function, I tried adding a 1s timeout to the `ReplaySubject` `token`, which logs out a user successfully.

Reply:
The subject has no expiration because there was just never a case for it to have one previously.

Rather than timing it out, you could probably do `this.token.next(null)` or `this.token.next('')` when you call `reset()`. You might need to change the type of the subject to `ReplaySubject<string | undefined>` if you want to null it out. This way, any component that calls `context.api.getToken().subscribe()` after a reset/logout will immediately receive a null/empty string value.

You can prevent components from getting the null/empty string and have them instead receive nothing at all until a valid token is available pretty easily:

```typescript
context.api.getToken()
  .pipe(filter(t => !!t)) // !! in JavaScript gives the "truthiness" of a value - null is falsy and empty string are falsy. Nonempty strings are truthy
  .subscribe( ... )
```